multichain-testing/tools/agd-lib.js

```diff
@@ -93,16 +101,122 @@ export const makeAgd = ({ execFileSync }) => {
 
     const outJson = flags({ output: 'json' });
 
+    /** @type {Record<string, any> | undefined} */
+    let version;
+
+    /** @type {((ev: EventQuery | EventQuery[]) => string[]) | undefined} */
+    let buildEventQueryArgs;
+
     const ro = freeze({
       status: async () => JSON.parse(exec([...nodeArgs, 'status'])),
+      version: async () => {
+        if (version) {
+          return version;
+        }
+
+        // TODO: This nonsense is because some appds write version to stderr!
```

Comment:
🤦 

could this be queried over RPC? If so, not a blocker but a good TODO 

the current line is more of an XXX: it's acknowledging a hack but not saying what to do